src/runtime/google-analytics-event-listener-test.js

```diff
@@ -28,21 +28,28 @@ import {analyticsEventToGoogleAnalyticsEvent} from './event-type-mapping';
 describes.realWin('GoogleAnalyticsEventListener', {}, (env) => {
   let win;
   let winMock;
+  let dataLayerMock;
   let eventManager;
   let deps;
   let listener;
 
   beforeEach(() => {
     sandbox.stub(self.console, 'log');
+    dataLayerMock = sandbox.mock({
```

Comment:
Could you add `dataLayerMock.verify()` to the `afterEach` method? Without this, the tests won't verify the mock's expectations

Reply:
I'd prefer any assertions to go into the individual tests rather than `afterEach`. This is widespread in the codebase but it's an anti-pattern to perform assertions in setup/teardown code.

Reply:
I view calling `verify()` on a mock different;y than defining a specific expectation within a test

Reply:
For instance, this expectation never gets checked if `verify()` isn't called on the mock
![image](https://user-images.githubusercontent.com/1216762/206567644-48e168e9-0478-477c-8903-9e1493079351.png)

Reply:
In that case, a specific test creates the expectation

Reply:
I'm wondering if it's worth every test setting expectations on mocks to also call `.verify()` on the mock... perhaps?

Reply:
In general I would prefer calling `verify` in each test. This has some advantages:

- Highlights the specific verification being performed in the test.
- Tests look more like arrange, act, assert instead of assert, arrange, act.
- The test runner will report failed expectations as a failure in the test case instead of an error in afterEach.

Given the style of the mock API we're using this may not always be practical, or could cause broken tests to pass because the author forgets to add `verify`. So, I guess I'd leave it up to the author.

Reply:
Good points, I'm liking this idea more now

Reply:
Makes sense. Reorganized the tests to have `verify` where it's needed.